src/main/java/com/labsynch/labseer/service/ParentSwapStructuresServiceImpl.java

```diff
@@ -29,6 +32,36 @@ public class ParentSwapStructuresServiceImpl implements ParentSwapStructuresServ
 	@Autowired
 	public ChemStructureService chemStructureService;
 
+	/**
+	 * Get parent matching the name. If no parent is found then check if the name
+	 * matches a unique alias and if a match is found then return the parent.
+	 *
+	 * @param name Parent corporate name or parent alias name.
+	 * @return   Parent object.
+	 */
+	private Parent getParent(String name) throws ParentNotFoundException {
+		// Check if a parent with the corporate name exists.
+		try {
+			return Parent.findParentsByCorpNameEquals(name).getSingleResult();
+		} catch (Exception e) {
+			logger.warn("No parent found for {}", name);
+		}
+
+		// Check if a unique alias with the name exists.
+		List<ParentAlias> parentAliases = new ArrayList<>();
+		try {
+			parentAliases = ParentAlias.findParentAliasesByAliasNameEquals(name).getResultList();
+		} catch (Exception e) {
+			logger.warn("No parent aliases found for {}", name);
+		}
+		if (parentAliases.size() == 1) {
+			ParentAlias parentAlias = parentAliases.get(0);
+			logger.info("Unique parent alias found for {}",  name);
+			return parentAlias.getParent();
+		}
+		throw new ParentNotFoundException("No parent or unique alias found for " + name);
```

Comment:
I think it'd actually be better to split this out and throw a different message depending on whether `parentAliases.size()` is 0 or > 1. You have the information at your fingertips to give a more complete message, for example providing the corporate IDs if there are multiple duplicate aliases (size>1), or communicating a clearer message that the name was not found as a corporate id OR an alias (size=0)

Reply:
Yep, I agree. I have updated the error and log messages. Let me know if the messages are appropriate now!